semgrep-core/tests/java/cp_try_return.java

```diff
@@ -0,0 +1,44 @@
+// https://github.com/returntocorp/semgrep/issues/4091
+public class Main {
+    public boolean test1() {
+        Throwable throwable = null;
+        while (cond) {
+            try {
+                return could_throw();
+            }
+            catch (Throwable t) {
+                throwable = t;
+            }
+        }
+        //OK:
+        return throwable == null;
+    }
+
+    public boolean test2() {
+        int cannot_throw = 42;
+        Throwable throwable = null;
+        while (cond) {
+            try {
+                return cannot_throw;
+            }
+            catch (Throwable t) {
+                // dead code
+                throwable = t;
+            }
+        }
+        //ERROR:
+        return throwable == null;
+    }
+
+   public boolean test3() {
```

Comment:
seems quite similar to test1
Not sure what are the consequences of the extra while(cond) in test1() on the CFG problem of this PR.

Reply:
I added some comments to the test case to explain what we are trying to test in each case. But `test3` in particular is just there because it was in the bug report, it is not particularly useful in this case. I added a more meaningful test based on the same pattern in tainting_rules/try_return.